### PR TITLE
[firefly-iii] Added firefly-iii chart

### PR DESCRIPTION
**Pull request by bikram990**
**Description of the change**

Add firefly-iii chart

**Benefits**

People will be able to deploy firefly-iii and manage their personal finance.

**Possible drawbacks**

NA

**Applicable issues**

NA

**Additional information**

**Checklist** 
- [X] Title of the PR starts with chart name (e.g. `[home-assistant]`)
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Chart `artifacthub.io/changes` changelog annotation has been updated in `Chart.yaml`. See [Artifact Hub documentation](https://artifacthub.io/docs/topics/annotations/helm/#supported-annotations) for more info.
- [x] Variables have been documented in the `values.yaml` file.

### CODE REVIEW

**Comment by Truxnell** [comment]
Hi @bikram990, thanks for the PR
Looks like you have some whitespace/end of file linting to do.  If you haven't, we suggest you install the repo pre-commit to ensure your PR will past the linting check.

**Comment by bikram990** [comment]
@Truxnell I've fixed the pre-commit linting issue.

File: charts/stable/firefly-iii/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+apiVersion: v2
+appVersion: 5.6.14
```
Comment (Truxnell) [inline charts/stable/firefly-iii/Chart.yaml:3]:
```suggestion
appVersion: version-5.6.14
```
We recommend putting the image tag as the appVersion and leaving the tag blank in the values, as it inherits this line.

File: charts/stable/firefly-iii/values.yaml [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+  tag: version-5.6.14
+
+strategy:
+  type: Recreate
+
+cronjobs:
+  # Enable cronjob for creating recurring transactions
+  recurring:
+    enabled: false
+    schedule: "0 3 * * *"
```
Comment (Truxnell) [inline charts/stable/firefly-iii/values.yaml:23]:
```suggestion
    schedule: "0 3 * * *" # Schedule time in UTC
```
Worth mentioning to end users the cron will be scheduled in UTC time?

File: charts/stable/firefly-iii/values.yaml [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+# https://github.com/k8s-at-home/library-charts/tree/main/charts/stable/common/values.yaml
+#
+
+image:
+  # -- image repository
+  repository: fireflyiii/core
+  # -- image pull policy
+  pullPolicy: IfNotPresent
+  # -- image tag
+  tag: version-5.6.14
```
Comment (Truxnell) [inline charts/stable/firefly-iii/values.yaml:14]:
```suggestion
  # @default -- chart.appVersion
  tag:
```
These changes, coupled with the adjustment in chart.yaml will allow maintaing the image tag in one place not two

File: charts/stable/firefly-iii/values.yaml [hunk 4]
```diff
@@ -0,0 +1,462 @@
...
+
+strategy:
+  type: Recreate
+
+cronjobs:
+  # Enable cronjob for creating recurring transactions
```
Comment (Truxnell) [inline charts/stable/firefly-iii/values.yaml:20]:
```suggestion
  # -- Enable cronjob for creating recurring transactions
```
Any reason you havent gone with the `# --` style for all comments?  These allow the comments to be detected as field comments & render in helm-docs.  All your comments in the file are without the `--` which would generate a fairly empty doc.
(Ref: https://github.com/norwoodj/helm-docs)

File: charts/stable/firefly-iii/values.yaml [hunk 5]
```diff
@@ -0,0 +1,462 @@
...
+            service:
+              port: http
+
+# -- Configure persistence settings for the chart under this key.
+# @default -- See values.yaml
+persistence:
+  config:
+    enabled: false
+  uploads:
+    enabled: true
```
Comment (Truxnell) [inline charts/stable/firefly-iii/values.yaml:419]:
```suggestion
    enabled: false
```
We usually leave persistance false as default and allow the users to chose to enable.

**Review by Truxnell — CHANGES_REQUESTED** [review]
Just a few changes to suggest
Also note pinned issue https://github.com/k8s-at-home/charts/issues/1572 regarding planned future deprecation of database dependencies that we have upcoming in the common-library chart.  It may not affect this chart but its worth noting the maintenance issues that the extra database dependencies can cause maintainers.

**Review by Truxnell — APPROVED** [review]

**Comment by Truxnell** [comment]
@bikram990 I've approved & merged.  
Have a look at how the readme generates - you can clean up anything that might need updating in a update (as it usually takes a bit to shake out changes with a new chart)

**Comment by Truxnell** [comment]
@all-contributors please add @bikram990  for code

**Pull request merged**